tests/main/buildmode/task.yaml

```diff
@@ -0,0 +1,22 @@
+summary: Test that we use the right buildmode per arch
+
+systems: [ubuntu-1*, ubuntu-2*]
+
+execute: |
+    # `objdump -f` will report "DYNAMIC" for binaries build with
+    # -buildmode=pie
+    needle="DYNAMIC"
+    
+    # armhf/i386 build without -buildmode=pie as it breaks too many
+    # things if we do.
+    if [ "$(uname -m)" = armv7l ] || [ "$(uname -m)" = i686 ]; then
```

Comment:
Could you instead make a test that checks for 32bit architecture? I know it's not quite the same but I trust that more than the armv7l vs armwhateveracronymoftheday or i[3456]86 encoding.

Reply:
Perhaps just run "file" on a binary and look for "ELF 32-bit", if it matches perform the pie-or-not test?

Reply:
Thanks, nice idea! I did something similar now, using "getconf LONG_BIT" to avoid looking at the machine name.